client/shipping-label/views/purchase/steps/rates/test/list.js

```diff
@@ -0,0 +1,384 @@
+				expect( errorWrapper.prop( 'text' ) ).to.equal( 'There was an error!' );
+			} );
+			it( 'should not display extra heading for package title', () => {
+				// the package title is displayed using the dropdown legend
+				expect( wrapper.find( 'p' ) ).to.have.length( 0 );
+			} );
+		} );
+
+		describe( 'Both rates and errors', () => {
+			const wrapper = shallow(
+				<ShippingRates
+					{ ...singlePackageProps }
+					availableRates={ {
+						weight_0_individual: Object.assign( {}, package_1_rate, package_1_error ),
+					} }
+					selectedRates={ {
+						weight_0_individual: '',
+					} }
+					errors={ {
+						server: {
+							weight_0_individual: [ {
+								message: 'There was an error!',
+							} ],
+						},
+						form: {
+							weight_0_individual: null,
+						},
+					} }
+				/>
+			);
+			it( 'should display rates drop down', () => {
+				expect( wrapper.find( Dropdown ) ).to.have.length( 1 );
+			} );
+			it( 'should display error', () => {
+				expect( wrapper.find( FieldError ) ).to.have.length( 1 );
+			} );
+			it( 'should not display extra heading for package title', () => {
+				// the package title is displayed using the dropdown legend
+				expect( wrapper.find( 'p' ) ).to.have.length( 0 );
+			} );
+		} );
+	} );
+
+	describe( 'Multiple packages', () => {
+		describe( 'Rates only for all packages', () => {
+			const wrapper = shallow(
+				<ShippingRates
+					{ ...multiPackageProps }
+					availableRates={ {
+						weight_0_individual: package_1_rate,
+						weight_1_individual: package_2_rate,
+					} }
+					errors={ {} }
+				/>
+			);
+			it( 'should display dropdown for each package', () => {
+				expect( wrapper.find( Dropdown ) ).to.have.length( 2 );
+			} );
+			it( 'should not display any errors', () => {
+				expect( wrapper.find( FieldError ) ).to.have.length( 0 );
+			} );
+			it( 'should not display extra heading for package title', () => {
+				// the package title is displayed using the dropdown legend
+				expect( wrapper.find( 'p' ) ).to.have.length( 0 );
+			} );
+		} );
+
+		describe( 'Errors only for all packages', () => {
+			const wrapper = shallow(
+				<ShippingRates
+					{ ...multiPackageProps }
+					availableRates={ {
+						weight_0_individual: package_1_error,
+						weight_1_individual: package_2_error,
+					} }
+				/>
+			);
+			it( 'should not display any dropdowns', () => {
+				expect( wrapper.find( Dropdown ) ).to.have.length( 0 );
+			} );
+			it( 'should display error for each package', () => {
+				expect( wrapper.find( FieldError ) ).to.have.length( 2 );
+			} );
+			it( 'should display extra heading for package title', () => {
+				const packageNames = wrapper.find( 'p' ).map( node => node.text() );
+				expect( wrapper.find( 'p' ) ).to.have.length( 2 );
+				expect( packageNames ).to.eql( [ '#1 - Package 1 Name', '#2 - Package 2 Name' ] );
+			} );
+		} );
+
+		describe( 'Both rates and errors for all packages', () => {
```

Comment:
Sure, I can add it, but it seems a bit superfluous to me. We're already testing the display of these cases:
- multiple packages
- errors only for each package
- rates only for each package
- both rates and errors for a package

Since we are capable of displaying multiple packages, and for each package we are capable of displaying each type: both errors and rates, rates only, and errors only, then in terms of display, we're all set.

What do you think?

In any case, I did some of the groundwork and it's available in 8c0b08ef000df3dd28e3d97842402e87fd04307c